lib-ee/emqx_ee_bridge/test/emqx_ee_bridge_mongodb_SUITE.erl

```diff
@@ -151,6 +151,7 @@ mongo_config(MongoHost, MongoPort0, rs = Type) ->
             "  servers = [~p]\n"
             "  w_mode = safe\n"
             "  database = mqtt\n"
+            "  mongo_type = rs\n"
```

Comment:
It's a bit strange to have to specify this explicitly, since the bridge type already defines the mongo type unambiguously.  Maybe we could keep the default value in the schema rather than the `required => true` property?

Reply:
The bridge type name is “mongodb” but not “mongo_rs”.
“mongo_rs” is the type name, but not the value.

Reply:
The bridge type is [`mongodb_rs`](https://github.com/emqx/emqx/pull/9653/files/fa1e605a6de5073f5daab38defe707af6e620fe1#diff-0df340b97fc1818f6dc7eca51a7991d0abf6894e39871c7c5f7f114e7982ac9fR147), which implies the `mongo_type` value.

Reply:
The schema previously had “default” which makes the “required” field meaningless. I believe the intention was to define that it’s required, but default was a mistake

Reply:
If the bridge type didn't already imply the mongo type unambiguously, then yes, I agree that the `mongo_type` should be required without a default.  E.g.: if the bridge type was just `mongo`, then the user would have to define if `mongo_type = rs | single | sharded`.

But, in this case, the is `mongodb_rs`, which _only_ accepts `mongo_type = rs`.  So , the latter becomes redundant.

Reply:
technically there is no need for this sub-id at all because the union members had different sets of fields,
so the type check can actually tell which exact member it is.

however the `mongo_type` field was actually intended for the 'identity' or 'signature' of the union member
it was just not made use of until now.

if we are to design the union selector using other field name(s) as signature, then it's as good as no selector at all.

so the field is redundant, but it helps to make the type check error messages more readable.

Reply:
Unless I'm understanding things wrong, I believe that [`mongodb_rs`](https://github.com/emqx/emqx/blob/22707495aca4cc1ed4d83451f1179a0b72397bdd/lib-ee/emqx_ee_bridge/src/emqx_ee_bridge.erl#L114) is the _bridge type_, and the _bridge name_ is [`?MODULE`](https://github.com/emqx/emqx/pull/9653/files#diff-0df340b97fc1818f6dc7eca51a7991d0abf6894e39871c7c5f7f114e7982ac9fR144), which indeed is arbitrary.

The types, AFAIK, are defined here: https://github.com/emqx/emqx/blob/22707495aca4cc1ed4d83451f1179a0b72397bdd/lib-ee/emqx_ee_bridge/src/emqx_ee_bridge.erl#L114

Reply:
my bad, fixing the union selector to not rely on mongo_type.

Reply:
changed the filed schema `required => false`, but the union type selector will fail if the value is not provided.